subprojects/core/src/main/java/org/gradle/execution/plan/TaskInAnotherBuild.java

```diff
@@ -209,6 +205,17 @@ public Path getTaskIdentityPath() {
         public String getTaskPath() {
             return task.getPath();
         }
+
+        @Override
+        public void resolveDependencies(TaskDependencyResolver dependencyResolver, Action<Node> processHardSuccessor) {
+            Set<Node> dependencies = dependencyResolver.resolveDependenciesFor(task, task.getTaskDependencies());
+            for (Node targetNode : dependencies) {
+                if (targetNode instanceof TaskNode) {
+                    addDependencySuccessor(targetNode);
+                    processHardSuccessor.execute(targetNode);
+                }
+            }
+        }
```

Comment:
We must also compute the dependencies of this foreign task so that we can construct the full task graph.

The instanceof check here guards from including other types of nodes, such as transforms that we currently don't distinguish whether they belong to this build or another. Without this check, such nodes from foreign builds would be added to the current build's execution plan.
I am not sure this is entirely correct now.

We might have to add other types of dependencies here like it is done in `LocalTaskNode`: https://github.com/gradle/gradle/blob/0e0220d40a93d6dfce0d7f0638662144e4e499f0/subprojects/core/src/main/java/org/gradle/execution/plan/LocalTaskNode.java#L132-L144